vortex-array/src/expr/exprs/cast.rs

```diff
@@ -86,32 +87,32 @@ impl VTable for Cast {
         })
     }
 
-    fn stat_max(
+    fn stat_expression(
         &self,
         expr: &ExpressionView<Self>,
-        catalog: &mut dyn StatsCatalog,
+        stat: Stat,
+        catalog: &dyn StatsCatalog,
     ) -> Option<Expression> {
-        expr.children()[0].stat_max(catalog)
-    }
-
-    fn stat_min(
-        &self,
-        expr: &ExpressionView<Self>,
-        catalog: &mut dyn StatsCatalog,
-    ) -> Option<Expression> {
-        expr.children()[0].stat_min(catalog)
-    }
-
-    fn stat_nan_count(
-        &self,
-        expr: &ExpressionView<Self>,
-        catalog: &mut dyn StatsCatalog,
-    ) -> Option<Expression> {
-        expr.children()[0].stat_nan_count(catalog)
-    }
-
-    fn stat_field_path(&self, expr: &ExpressionView<Self>) -> Option<FieldPath> {
-        expr.children()[0].stat_field_path()
+        match stat {
+            Stat::IsConstant
+            | Stat::IsSorted
+            | Stat::IsStrictSorted
+            | Stat::Max
+            | Stat::Min
+            | Stat::NaNCount
+            | Stat::Sum
+            | Stat::UncompressedSizeInBytes => expr.child(0).stat_expression(stat, catalog),
+            Stat::NullCount => {
+                // if !expr.data().is_nullable() {
+                // NOTE(ngates): we should decide on the semantics here. In theory, the null
+                //  count of something cast to non-nullable will be zero. But if we return
+                //  that we know this to be zero, then a pruning predicate may eliminate data
+                //  that would otherwise have caused the cast to error.
+                // return Some(lit(0u64));
+                // }
+                None
```

Comment:
I think we should move error into the pruning predicate, can assert that child `null_count == 0`